src/coreComponents/physicsSolvers/SolverBase.hpp

```diff
@@ -777,6 +782,20 @@ string SolverBase::getConstitutiveName( ElementSubRegionBase const & subRegion )
   return validName;
 }
 
+template< typename CONSTITUTIVE_BASE_TYPE >
+string SolverBase::getConstitutiveName( ParticleSubRegionBase const & subRegion ) // particle overload
```

Comment:
We should be able to build up a proper class architecture such that we do not have to duplicate to much code and patterns.

Reply:
It's a bit tricky because particles have connections to constitutive data (like elements) but also have velocity, acceleration, etc. (like nodal data).  We'd discussed ideas of having particle be some special kind of single-node element but the existing finite element data structure is not ideal for the MPM algorithm.

Having a constitutive model be something that could be attached to any object (an element, a gauss point, a node, a particle, etc.) might be a good direction to go.